### DISCUSSION TITLE
Find attributes in resource from extends

### DISCUSSION THREAD

**Discussion by valentinrougier39** [dedoc/scramble#437] [Ideas]
Hi, I use resources which are extended, when project Resource is extended but not overloaded, I don't have attributes in doc. 
Exemple : I have a DogResource in my project but I extend this class from a package with DogPackageResource in DogPackageResource the method toArray() is ready and If I don't overload it, the doc is empty.
Is a possibility of improvement ? 
Thank you

**Comment by romalytvynenko** [comment]
@valentinrougier39 I think I get the question, but still if you share some code examples I'd appreciate it

**Reply by valentinrougier39** [reply]
Here are 3 screenshots, you can see both resources and the preview in Scramble !

<img width="430" alt="Capture d’écran 2024-06-28 à 19 18 32" src="https://github.com/dedoc/scramble/assets/62744879/d4d8b42a-d367-4dc1-a4a7-4fae6d7280ed">
<img width="1277" alt="Capture d’écran 2024-06-28 à 19 21 18" src="https://github.com/dedoc/scramble/assets/62744879/4a9b341a-e151-4121-b9ab-6c23fcd239f0">
<img width="481" alt="Capture d’écran 2024-06-28 à 19 22 59" src="https://github.com/dedoc/scramble/assets/62744879/529c2453-1459-458e-bf4b-3d0d02437ed8">

**Reply by romalytvynenko** [reply]
@valentinrougier39 ah, gotcha. That totally makes sense.

**Reply by romalytvynenko** [reply]
@valentinrougier39 in your project this resource class you're extending is located in vendor folder? Or this is your project's file?

**Reply by valentinrougier39** [reply]
@romalytvynenko DogResource is in the project yes, DogPackageResource is in vendor folder

**Reply by valentinrougier39** [reply]
Hi @romalytvynenko , do you think you will plan this as an improvement soon ? 
Thank you

**Reply by romalytvynenko** [reply]
@valentinrougier39 It definitely is, I'm just thinking how to enable classes analysis that are in vendor. For now Scramble skips them to make sure it is fast.

**Reply by valentinrougier39** [reply]
Ok nice, good luck :)

**Reply by valentinrougier39** [reply]
Hi, do you have some news about the feature ?

**Reply by valentinrougier39** [reply]
Hi @romalytvynenko , what's about this ask ? Thank you

**Reply by romalytvynenko** [reply]
Hey hey! Added the issue: https://github.com/dedoc/scramble/issues/505

**Reply by romalytvynenko** [reply]
@valentinrougier39 To be honest, after trying it out, I don't think this really fits into the core, so here is the extension you can try: 

```php
<?php

namespace Dedoc\Scramble\Support\InferExtensions;

use Dedoc\Scramble\Infer;
use Dedoc\Scramble\Infer\Extensions\AfterClassDefinitionCreatedExtension;
use Dedoc\Scramble\Infer\Extensions\Event\ClassDefinitionCreatedEvent;
use Illuminate\Http\Resources\Json\JsonResource;
use Illuminate\Support\Str;

class AnalyzeVendorJsonResourcesExtension implements AfterClassDefinitionCreatedExtension
{
    public function shouldHandle(string $name): bool
    {
        return is_a($name, JsonResource::class, true) && $name !== JsonResource::class;
    }

    public function afterClassDefinitionCreated(ClassDefinitionCreatedEvent $event)
    {
        if (! $this->isClassInVendor($event->name)) {
            return;
        }

        app(Infer::class)->analyzeClass($event->name);
    }

    private function isClassInVendor(string $className): bool
    {
        $filePath = (new \ReflectionClass($className))->getFileName();

        return Str::contains($filePath, DIRECTORY_SEPARATOR.'vendor'.DIRECTORY_SEPARATOR);
    }
}

```

**Reply by romalytvynenko** [reply]
It works by enforcing analysis of json resource files from vendor, after they've been skipped due to being a vendor files

**Reply by romalytvynenko** [reply]
I may reconsider this and add it to core, if you share the package you use (that one that has JsonResources in it). I'm simply not sure that analyzing that files will solve the issue – maybe they have some magic in `toArray`, I don't know.

**Reply by valentinrougier39** [reply]
Hi @romalytvynenko , thank you, I'll try this soon
